src/Stepper/StepperHeaderStep.jsx

```diff
@@ -7,28 +7,36 @@ import { StepperContext } from './StepperContext';
 import { Icon, Bubble } from '..';
 
 function StepperHeaderStep({
-  eventKey,
   title,
   isActive,
   hasError,
   description,
   index,
+  onClick,
 }) {
-  const { getIsComplete } = useContext(StepperContext);
-  const isComplete = getIsComplete(eventKey);
+  const { getIsViewed } = useContext(StepperContext);
+  const isComplete = getIsViewed(index + 1);
+  const isViewed = getIsViewed(index);
   const stepIcon = isComplete ? <Icon src={Check} /> : <span>{index + 1}</span>;
   const errorIcon = <Icon src={Error} />;
+  const isClickable = onClick && isViewed && !isActive;
 
   return (
     <li
+      /* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */
+      tabIndex={isClickable ? 0 : -1}
+      role="presentation"
```

Comment:
Yes, you are right

Reply:
I think eslint just looks at the `li` tag (and doesn't pay much attention to role) and basically says that `li` tag is non-interactive by definition no matter the role. Actually if you just plainly add `role="button"` eslint starts complaining with `Non-interactive elements should not be assigned interactive roles.(jsx-a11y/no-noninteractive-element-to-interactive-role)`.

I don't see a workaround here, I think if step is clickable we need to just use `button` tag instead of `li` to comply with `a11y` rules

Reply:
@PKulkoRaccoonGang @viktorrusakov Thanks for the clarifications! I think it's worth trying to make use of `button` when it's clickable in this case to avoid needing to ignore a11y rules, given a11y is core to Paragon.